media-libs/amdgpu-pro-vulkan/amdgpu-pro-vulkan-21.30.1290604.ebuild

```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MULTILIB_COMPAT=( abi_x86_{32,64} )
+
+inherit unpacker multilib-minimal
+
+MY_PV_MAJOR=`ver_cut 1`
+MY_PV_MINOR=`ver_cut 2`
+MY_PV_REV=`ver_cut 3`
+MY_PV="${MY_PV_MAJOR}.${MY_PV_MINOR}-${MY_PV_REV}"
+
+DESCRIPTION="AMD's closed source vulkan driver, from Radeon Software for Linux"
+HOMEPAGE="https://www.amd.com/en/support/kb/release-notes/rn-amdgpu-unified-linux-21-30"
+SRC_URI="https://drivers.amd.com/drivers/linux/amdgpu-pro-${MY_PV}-ubuntu-20.04.tar.xz"
+S="${WORKDIR}"
+
+RESTRICT="bindist mirror fetch strip"
+
+LICENSE="AMD-GPU-PRO-EULA"
+SLOT="0"
+KEYWORDS="-* ~amd64 ~x86"
+
+QA_PREBUILT="
+	/usr/lib64/amdvlkpro64.so*
+	/usr/lib/amdvlkpro32.so*
+"
+
+pkg_nofetch() {
+	einfo "Please download:"
+	einfo "    Radeon™ Software for Linux® version ${MY_PV_MAJOR}.${MY_PV_MINOR} for Ubuntu 20.04.3"
+	einfo "from ${HOMEPAGE} and place them in your DISTDIR directory."
+}
+
+src_unpack() {
+	default
+
+	if usex "abi_x86_64" ; then
+		mkdir "${S}/vulkan-amdgpu-pro-amd64" || die
+		cd "${S}/vulkan-amdgpu-pro-amd64" || die
+		unpack_deb "${S}/amdgpu-pro-${MY_PV}-ubuntu-20.04/vulkan-amdgpu-pro_${MY_PV}_amd64.deb"
+	fi
+
+	if usex "abi_x86_32" ; then
+		mkdir "${S}/vulkan-amdgpu-pro-i386" || die
+		cd "${S}/vulkan-amdgpu-pro-i386" || die
+		unpack_deb "${S}/amdgpu-pro-${MY_PV}-ubuntu-20.04/vulkan-amdgpu-pro_${MY_PV}_i386.deb"
+	fi
+}
+
+src_prepare() {
+
+	if usex "abi_x86_64" ; then
+		cd "${S}/vulkan-amdgpu-pro-amd64/opt/amdgpu-pro/lib/x86_64-linux-gnu/" || die
+		mv "amdvlk64.so.1.0" "amdvlkpro64.so.1.0" || die
+
+		cd "${S}/vulkan-amdgpu-pro-amd64/opt/amdgpu-pro/etc/vulkan/icd.d/" || die
+		eapply -p0 "${FILESDIR}/icd_amd64.patch"
+		mv "amd_icd64.json" "amd_pro_icd64.json" || die
+	fi
+
+	if usex "abi_x86_32" ; then
+		cd "${S}/vulkan-amdgpu-pro-i386/opt/amdgpu-pro/lib/i386-linux-gnu/" || die
+		mv "amdvlk32.so.1.0" "amdvlkpro32.so.1.0" || die
+
+		cd "${S}/vulkan-amdgpu-pro-i386/opt/amdgpu-pro/etc/vulkan/icd.d/" || die
+		eapply -p0 "${FILESDIR}/icd_x86.patch"
+		mv "amd_icd32.json" "amd_pro_icd32.json" || die
+	fi
+
+	eapply_user
+}
+
+src_install() {
+
+	if usex "abi_x86_64" ; then
+		insinto "/usr/lib64"
```

Comment:
I would still do $(get_libdir) for these.

Reply:
How does it know which lib to return ? :thinking:

Reply:
doh, it's not an 'else', yeah, that's fine as-is unless we do multilib and I don't know if we want to do that right now (head's not clear enough for it)

Reply:
I believe this ebuild is already multilib as emerge shows 
```shell
Calculating dependencies... done!
[ebuild   Rf   ] media-libs/amdgpu-pro-vulkan-21.30.1290604::gentoo-adel  ABI_X86="32 (64)" 0 KiB
```
But I have no idea if I am doing this right, feedback welcome! (when you will feel more fresh x) )